DigitalLearningSolutions.Web/Helpers/FilteringHelper.cs

```diff
@@ -20,70 +20,106 @@ public static class FilteringHelper
         {
             var listOfFilters = NewlineSeparatedStringListHelper.SplitNewlineSeparatedList(filterBy);
 
-            foreach (var filter in listOfFilters)
-            {
-                var splitFilter = filter.Split(Separator);
-                var propertyName = splitFilter[0];
-                var propertyValueString = splitFilter[1];
-                var propertyType = typeof(T).GetProperty(propertyName)!.PropertyType;
-                var propertyValue = TypeDescriptor.GetConverter(propertyType).ConvertFromString(propertyValueString);
+            var appliedFilters = listOfFilters.Select(filter => new AppliedFilter(filter));
 
-                items = items.Where(propertyName, propertyValue);
+            foreach (var filterGroup in appliedFilters.GroupBy(a => a.Group))
+            {
+                var itemsToFilter = items;
+                var setOfFilteredLists = filterGroup.Select(
+                    af => FilterGroupItems(itemsToFilter, af.PropertyName, af.PropertyValue)
+                );
+                items = setOfFilteredLists.SelectMany(x => x).Distinct().AsQueryable();
             }
 
             return items;
         }
+
+        private static IQueryable<T> FilterGroupItems<T>(
+            IQueryable<T> items,
+            string propertyName,
+            string propertyValueString
+        )
+        {
+            var propertyType = typeof(T).GetProperty(propertyName)!.PropertyType;
+            var propertyValue = TypeDescriptor.GetConverter(propertyType).ConvertFromString(propertyValueString);
+            return items.Where(propertyName, propertyValue);
+        }
+
+        private class AppliedFilter

```

Comment:
Looking at this, I think this file will get quite big quite quickly. Could we move this class to models and/or separate out the filteroptions?